src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py

```diff
@@ -49,7 +48,7 @@ def generic_data_service_factory(service, name=None, key=None, connection_string
 
 
 def storage_client_factory(**_):
-    return get_mgmt_service_client(StorageManagementClient)
+    return get_mgmt_service_client(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS)
```

Comment:
@tjprescott I think @derekbekoe wants to separate provider and resource type. And storage is not a good example for that, but `MGMT_COMPUTE_CONTAINERSERVICE` or `MGMT_RESOURCES_LOCKS` will be more obvious. So I'm ok personaly to keep it consistent even if it's weird for simple service like storage